lib/3scale/backend/stats/delete_job_def.rb

```diff
@@ -0,0 +1,59 @@
+module ThreeScale
+  module Backend
+    module Stats
+      class DeleteJobDef
+        ATTRIBUTES = %i[service_id applications metrics users from to context_info].freeze
+        private_constant :ATTRIBUTES
+        attr_accessor(*ATTRIBUTES)
```

Comment:
Do we need to expose the setter methods?

Right now, it's possible to initialize a valid instance, modify it with an invalid value, and then call .run_async.

Reply:
yeah, that's why validation was done in `.run_async` in the beginning. I will implement read-only class. :+1: